Add elapsed time in pending status

bncer opened the pull request:
Closes [#371](https://github.com/rust-lang/bors/issues/371)

Display elapsed time

Currently it's not live, it shows within template.

Kobzol commented:
I think we can do that in several steps (multiple PRs). It is enough to just show the elapsed time in the first PR (this one). Thank you! I will take a look tomorrow.

Kobzol commented on templates/queue.html:
```diff
@@ -340,5 +386,40 @@ <h1>
+    function updateElapsedTimes() {
+        const now = Date.now();
+        document.querySelectorAll('[data-created-at]').forEach(td => {
+            const createdAtMs = parseInt(td.getAttribute('data-created-at'), 10);
```
```suggestion
            const createdAtMs = parseInt(td.dataset.createdAt, 10);
```

Kobzol commented on templates/queue.html:
```diff
@@ -114,12 +154,18 @@ <h1>
                 </td>
-                <td data-status="{{ crate::templates::status_text(pr) }}">
+                {% let pending_build = crate::templates::get_pending_build(pr) %}
+                <td class="status" data-status="{{ crate::templates::status_text(pr) }}"
```
Sorry, I meant the `title="foo"` attribute (https://www.geeksforgeeks.org/html/html-title-attribute/), not the `<title>` element :)

bncer commented:
> Could you share the code where that happens?

[in src/database/operations.rs in upsert_pull_request function](https://github.com/rust-lang/bors/blob/3bb5fa203bbd1b8138738371a7d67f646d7b5a5a/src/database/operations.rs#L130C1-L190C2)
and [src/database/operations.rs in update_mergeable_states_by_base_branch function](https://github.com/rust-lang/bors/blob/3bb5fa203bbd1b8138738371a7d67f646d7b5a5a/src/database/operations.rs#L301C1-L352C11)

After several hours I built bors and it worked without any errors. But I tried to reproduce this by: `cargo clean`, `rm -rf .sqlx` and it fails again 😆

As a result `cargo sqlx prepare -- --all-targets`  shows modifications on:
modified:   .sqlx/query-367bf2de86a8a92b56b2be296cef119a7835ab5fd09326265a4a1403d0d6ad19.json
modified:   .sqlx/query-d16ca5631fcdd4216b1bf3a7b1672b8721d72dead28356fbbc69f84029d08132.json
```
"nullable": [
-      false,
-      false,
-      false,
-      false,
-      false,
-      false,
+      true,
+      true,
+      true,
+      true,
+      true,
+      true, ...
```

bncer commented:
With sqlx's exclamation mark (!) I lowered number of errors, but clear reason of different behavior blows my mind 
```
### diff --git a/src/database/operations.rs b/src/database/operations.rs
index 6d31e9b..bce49ee 100644
--- a/src/database/operations.rs
+++ b/src/database/operations.rs
@@ -151,22 +151,22 @@ pub(crate) async fn upsert_pull_request(
             )
             SELECT
                 pr.id,
-                pr.repository as "repository: GithubRepoName",
+                pr.repository as "repository!: GithubRepoName",
                 pr.number as "number!: i64",
                 pr.title,
                 pr.author,
-                pr.assignees as "assignees: Assignees",
+                pr.assignees as "assignees!: Assignees",
                 (
                     pr.approved_by,
                     pr.approved_sha
                 ) AS "approval_status!: ApprovalStatus",
-                pr.status as "pr_status: PullRequestStatus",
+                pr.status as "pr_status!: PullRequestStatus",
                 pr.priority,
-                pr.rollup as "rollup: RollupMode",
-                pr.delegated_permission as "delegated_permission: DelegatedPermission",
+                pr.rollup as "rollup!: RollupMode",
+                pr.delegated_permission as "delegated_permission!: DelegatedPermission",
                 pr.base_branch,
-                pr.mergeable_state as "mergeable_state: MergeableState",
-                pr.created_at as "created_at: DateTime<Utc>",
+                pr.mergeable_state as "mergeable_state!: MergeableState",
+                pr.created_at as "created_at!: DateTime<Utc>",
                 try_build AS "try_build: BuildModel",
                 auto_build AS "auto_build: BuildModel"
             FROM upserted_pr as pr
```
